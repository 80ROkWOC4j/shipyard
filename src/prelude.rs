pub use crate::get::GetComponent;
pub use crate::iter::{iterators, IntoIter, Shiperator};
pub use crate::not::Not;
pub use crate::remove::Remove;
pub use crate::run::System;
#[doc(hidden)]
pub use crate::run::SystemData;
pub use crate::sparse_set::{sort, sort::Sortable, View, ViewMut};
pub use crate::storage::{AllStorages, Entities, EntitiesMut, Key};
pub use crate::world::World;
pub use crate::Unique;
#[doc(hidden)]
#[cfg(feature = "proc")]
pub use shipyard_proc::system;

#[cfg(feature = "parallel")]
pub use crate::ThreadPool;
