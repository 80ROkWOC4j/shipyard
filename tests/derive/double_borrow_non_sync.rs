use shipyard::prelude::*;

struct NonSyncStruct(core::marker::PhantomData<*const ()>);

unsafe impl Send for NonSyncStruct {}

#[system(NonSyncSys)]
fn run(_: NonSync<&NonSyncStruct>, _: NonSync<&mut NonSyncStruct>) {}

#[system(NonSyncSysUnique)]
fn run(_: NonSync<&NonSyncStruct>, _: Unique<NonSync<&mut NonSyncStruct>>) {}

#[system(UniqueNonSyncSys)]
fn run(_: Unique<NonSync<&NonSyncStruct>>, _: NonSync<&mut NonSyncStruct>) {}

#[system(UniqueUnique)]
fn run(_: Unique<NonSync<&NonSyncStruct>>, _: Unique<NonSync<&mut NonSyncStruct>>) {}

fn main() {}
