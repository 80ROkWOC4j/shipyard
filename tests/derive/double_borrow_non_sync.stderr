error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/double_borrow_non_sync.rs:8:8
  |
8 | fn run(_: NonSync<&NonSyncStruct>, _: NonSync<&mut NonSyncStruct>) {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow_non_sync.rs:11:8
   |
11 | fn run(_: NonSync<&NonSyncStruct>, _: Unique<NonSync<&mut NonSyncStruct>>) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow_non_sync.rs:14:8
   |
14 | fn run(_: Unique<NonSync<&NonSyncStruct>>, _: NonSync<&mut NonSyncStruct>) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow_non_sync.rs:17:8
   |
17 | fn run(_: Unique<NonSync<&NonSyncStruct>>, _: Unique<NonSync<&mut NonSyncStruct>>) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
